### DISCUSSION TITLE
Dellete commits

### DISCUSSION THREAD

**Discussion by ThienVie** [community/community#28912] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi GitHub Community. There is a random commit, which I want to delete, but I don't know how to do it. Do you have any ideas how I can delete it, so anyone would not see this? It would be really helpful. By the way, I am sorry, if there are similar discussions, because I am not familiar with some words, yet.

**Comment by ihnatenkoo** [comment, marked as answer]
Hello, if you want to delete some recent commits, you can use git reset --hard `<hash new last commit in your project>` and git push --force for update your repository.
If you want to delete random commit you must use git rebase --interactive function, read more about it on the link
[https://git-scm.com/docs/git-rebase](url)
[https://git-scm.com/docs/git-reset](url)

**Comment by airtower-luna** [comment]
> Do you have any ideas how I can delete it, so anyone would not see this?

This sounds like the commit might contain some kind of sensitive data. In that case there are two parts:

1. First, you need to rewrite the history of the repository to remove (or change) the commit in question. [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) describes how to do that. Note that this will change *all* commit IDs after the removed (or changed) commit. This means anyone who already has a copy of your repository and tries to fetch updates will have to manually accept the modified history. :warning:
2. After rewriting history, you need to ensure the commit isn't cached by GitHub any more. See the "Fully removing the data from GitHub" section of the [Removing sensitive data from a repository](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository) documentation page. That page also describes some other ways to rewrite repository geared at removing file(s) throughout repository history. They all change commit IDs, though.
